### DISCUSSION TITLE
Test failure for git 2.30.1 and 2.34.1 due to GIT_EXEC_PATH

### DISCUSSION THREAD

**Discussion by typesanitizer** [arxanas/git-branchless#230] [General]
When running tests locally from `master` using `PATH_TO_GIT="$(which git)" cargo test`, I'm seeing a failure for `command::test_undo::test_git_bisect_produces_empty_event` where `bisect` is not found.

A minimal reproducer for this is:

```
GIT_EXEC_PATH=/usr/bin git bisect -h
```

This fails with

```
git: 'bisect' is not a git command. See 'git --help'
```

It only fails for `bisect`, not other subcommands, so it seems like a Git bug, not a `git-branchless` bug.

With git 2.25.1, the test doesn't call bisect, so the test passes. With git 2.30.1 (AppleGit-130) on macOS, or 2.34.1 on Linux from, the test fails. Which makes sense, because that test only calls `bisect` for 2.29 or newer.

I see that the CI has recent Git versions in the build matrix here: https://github.com/arxanas/git-branchless/blob/b343a84f9850ad612da7543da6be3557e659c359/.github/workflows/linux.yml#L21

It's unclear to me how this test is passing in CI but failing locally. I'm wondering if there is a difference between even-numbered versions (stable?) vs odd numbers (unstable?).

I'll try to investigate this further, but just putting it up here, in case anyone else hits the same issue.

**Comment by typesanitizer** [comment]
I built git 2.33.1 from source on Linux and it has this same issue when running `GIT_EXEC_PATH=/usr/bin git bisect -h` (and the tests). So it seems to be some other configuration difference between the CI and my local machines. 🤔

**Reply by arxanas** [reply]
If you built Git from source, you might want to set `GIT_EXEC_PATH` to the same directory the `git` executable lives in; I think there'll be a `git bisect` in there.

**Reply by typesanitizer** [reply]
I installed the locally built git to `/usr/bin` with `make install`, hence the same `GIT_EXEC_PATH=/usr/bin`.

**Reply by typesanitizer** [reply]
If I run `GIT_EXEC_PATH=. git bisect -h` (or `./git bisect -h`) in the build directory, that works. Similarly, after installation, `/usr/bin/git bisect -h` works.

**Reply by arxanas** [reply]
`GIT_EXEC_PATH` generally won't be a path like `/usr/bin`. For example, this is the path on my system:

```
$ /usr/bin/git --exec-path
/Library/Developer/CommandLineTools/usr/libexec/git-core
```

My guess is that `make install` installs the Git helper tools somewhere other than `/usr/bin`.

If you run `GIT_EXEC_PATH=.` in the build directory, then it works because the helper tools happen to have been built in that directory.
